apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -0,0 +1,288 @@
+  private ses: AWS.SES
+
+  constructor(
+    private readonly emailTemplatesService: SfApiEmailTemplatesService,
+    private readonly conProfilesService: ConProfilesService,
+    private readonly conMentorshipMatchesService: ConMentorshipMatchesService
+  ) {
+    this.ses = new AWS.SES(sesConfig)
+  }
+
+  async getDraftingConProfiles({ userType }: { userType: UserType }) {
+    if (!userType) {
+      throw new Error('User type is required')
+    }
+
+    const createdDate = new Date()
+    createdDate.setDate(createdDate.getDate() - 14)
+    createdDate.setHours(0, 0, 0, 0) // set time to 00:00
+
+    const endOfCreatedDate = new Date(createdDate)
+    endOfCreatedDate.setHours(23, 59, 59, 999) // set time to 23:59
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': userType,
+      Profile_Status__c: ConnectProfileStatus.DRAFTING_PROFILE,
+      CreatedDate: {
+        $gte: jsforce.SfDate.toDateTimeLiteral(createdDate),
+        $lte: jsforce.SfDate.toDateTimeLiteral(endOfCreatedDate),
+      },
+    })
+  }
+
+  async getApprovedMenteesByDaysAndLocation({
+    daysAgo,
+  }: {
+    daysAgo: '7d' | '14d'
+  }) {
+    const approvedDate = new Date()
+    approvedDate.setDate(approvedDate.getDate() - parseInt(daysAgo)) // Interestingly, parseInt is able to parse 7d and 14d to 7 and 14 respectively
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': UserType.MENTEE,
+      Profile_Status__c: ConnectProfileStatus.APPROVED,
+      Profile_First_Approved_At__c: {
+        $eq: jsforce.SfDate.toDateLiteral(approvedDate),
+      },
+    })
+  }
+
+  async getConProfilesWithThreeMonthsOldMentorshipMatches() {
+    const NinetyDaysAgo = new Date().setDate(new Date().getDate() - 90)
+
+    const threeMonthsOldMentorshipMatches =
+      await this.conMentorshipMatchesService.findAll({
+        Status__c: MentorshipMatchStatus.ACCEPTED,
+        Start_Date__c: {
+          $eq: jsforce.SfDate.toDateLiteral(NinetyDaysAgo),
+        },
+      })
+
+    const menteeIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.menteeId
+    )
+    const mentorIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.mentorId
+    )
+
+    if ([...menteeIds, ...mentorIds].length > 0) {
+      return await this.conProfilesService.findAll({
+        'Contact__r.Id': {
+          $in: [...menteeIds, ...mentorIds],
+        },
+      })
+    }
+
+    return []
+  }
+
+  async sendCompleteProfileReminder({
+    userType,
+    email,
+    firstName,
+  }: {
+    userType: UserType
+    email: string
+    firstName: string
+  }) {
+    if (!userType) {
+      throw new Error('User type is required')
+    }
+
+    const sfEmailTemplateName =
+      userType === UserType.MENTOR
+        ? 'Mentor - Profile Completion Reminder'
+        : userType === UserType.MENTEE
+        ? 'Mentee - Profile Completion Reminder'
+        : null
+
+    if (!sfEmailTemplateName) {
```

Comment:
Hmmm... same?

Reply:
@helloanil, I'd like to add my 5 cents about using the template name. 

We should use the `developerName` value of the email template instead of its `name` value. PO or any manager can update the `name` value anytime and the above won't work anymore.

You can get the `developerName` value by opening a particular template on SF and then going to the _Salesforce Inspector_ extension. There, you click `Show all data` and scroll until you find that template's `developerName` value. This value can also be changed in Salesforce Inspector, but only devs and SF admin know how to do it, and I am sure none of us would do it without a reason.

![image](https://github.com/talent-connect/connect/assets/51786805/500f627c-393f-49c6-bfd0-17b0cef2393b)